### DISCUSSION TITLE
tracing-subscriber: can't filter events by field

### DISCUSSION THREAD

**Discussion by stevenengler** [tokio-rs/tracing#2108] [Q&A]
### Version

```
tracing-test v0.1.0
├── tracing v0.1.34
│   ├── tracing-attributes v0.1.21 (proc-macro)
│   └── tracing-core v0.1.26
└── tracing-subscriber v0.3.11
    ├── tracing v0.1.34 (*)
    ├── tracing-core v0.1.26 (*)
    └── tracing-log v0.1.3
        └── tracing-core v0.1.26 (*)
```

### Platform

```
Linux hostname 5.13.0-40-generic #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Description

This seems like such a simple issue that I'm guessing I must be doing something wrong, but when I follow the documentation it's not working as I'd expect.

The [`EnvFilter`](https://docs.rs/tracing-subscriber/0.3.11/tracing_subscriber/filter/struct.EnvFilter.html#directives) documentation says that the `field` should match on events:

> `field` matches on fields within spans. Field names can also be supplied without a `value` and will match on any `Span` or `Event` that has a field with that name. For example: `[span{field=\"value\"}]=debug`, `[{field}]=trace`.

When I run the following code:

```rust
#[tracing::instrument]
fn test() {
    tracing::event!(tracing::Level::INFO, abc = 5, "In test fn");
}

fn main() {
    tracing_subscriber::fmt::init();
    test();
}
```

with:

```
$ RUST_LOG='tracing_test[test{abc}]=trace' cargo run
$ RUST_LOG='tracing_test[test{abc=5}]=trace' cargo run
```

The event is not printed (there is no output) in either case. The `test{abc}` filter doesn't work when the `EnvFilter` documentation seems to imply that it should.

If instead I run it without the field filter:

```
$ RUST_LOG='tracing_test[test]=trace' cargo run
```

The `INFO test: tracing_test: In test fn abc=5` line is printed.

**Comment by hawkw** [comment]
The directive `[test{abc}]` matches a _span_ whose name is `test` with a field named `abc`. To match an event with a field named `abc`, you would need to write just `[{abc}]`.

The match expressions `[test{abc=5}]` or `[{abc=5}]` will not match an event, because field value filters are currently only supported for spans.

Sorry if the documentation is unclear, if you have any suggestions for how we can make this clearer, that would be great.
